torch/jit/__init__.py

```diff
@@ -2013,6 +2013,21 @@ def _get_script_class(name):
 # torch.jit.Error
 Error = torch._C.JITException
 
+def _get_named_tuple_properties(obj):
+    assert issubclass(obj, tuple) and hasattr(obj, '_fields')
+    fields = list(obj._fields)
+    annotations = []
+    has_annotations = hasattr(obj, '__annotations__')
+    for field in fields:
+        if has_annotations and field in obj.__annotations__:
+            annotations.append(torch.jit.annotations.ann_to_type(obj.__annotations__[field]))
+        else:
+            annotations.append(TensorType.get())
+    return fields, annotations
+
+def _create_named_tuple(t, names):
+    TupleType = collections.namedtuple('NamedTuple', names)
```

Comment:
We should still be extracting the unqualified class name in `_get_named_tuple_properties`, putting it into TupleType and restoring it correctly here.